src/Orleans.Core/Configuration/Options/StorageOptions.cs

```diff
@@ -0,0 +1,13 @@
+
+namespace Orleans.Hosting
+{
+    public class StorageOptions
```

Comment:
I'm not comfortable with the approach where different storage provider uses the same storage options. The point of moving to ASP.NET is to move to the direction where different component has its own options, which fits it most. Having one option for all storage component is against that principle. 

For example, you used this option in DynamoDB reminder table. But in dynamo Db storage , there's no concept of data connection string. Instead of that, it has access key and secret key. While azure storage has the concept of data connection string. So a one for all solution is not good.

Reply:
I understand and agree, but this was how the previous configuration worked.  The host storage information was used for everything unless overridden by a component.  I do not agree with this pattern, but designing a new pattern was not part of this work.  Further, most of the existing providers make this same assumption, so we need to clean up all of that before we can move away from this pattern.  I have done some thought on how to address this, but am not addressing it in this PR.  I'll add it to the notes to be addressed in future passes.